### PR TITLE
feat(iceberg): use native glue impl for iceberg source

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- https://github.com/risingwavelabs/risingwave/issues/17548
- Use native glue impl for iceberg source.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by xxchan — APPROVED** [review]

File: src/connector/src/sink/iceberg/mod.rs [hunk 1]
```diff
@@ -553,15 +554,52 @@ impl IcebergConfig {
...
+                    .props(iceberg_configs);
+                let config = if let Some(uri) = self.uri.as_deref() {
+                    config_builder.uri(uri.to_string()).build()
+                } else {
+                    config_builder.build()
+                };
+                let catalog = iceberg_catalog_glue::GlueCatalog::new(config).await?;
+                Ok(Arc::new(catalog))
+            }
+            catalog_type if catalog_type == "hive" || catalog_type == "jdbc" => {
```
Comment (xxchan) [inline src/connector/src/sink/iceberg/mod.rs:597]: Are we going to change other types too later?

Comment (chenzl25) [reply src/connector/src/sink/iceberg/mod.rs:597]: Yes, I am trying to change the `hive` catalog later by `iceberg-catalog-hms`. But the `jdbc` one can't be changed with iceberg v0.3.0

**Comment by chenzl25** [comment]
meta panic after aws-config and aws-runtime upgrading

```
thread 'main' panicked at src/meta/node/src/server.rs:467:6:
called `Result::unwrap()` on an `Err` value: ObjectStore(s3 error: dispatch failure: other: UserAgentInterceptor modify_before_signing interceptor encountered an error: The UserAgentInterceptor requires ApiMetadata to be set before the request is made. This is a bug. Please file an issue.)
stack backtrace:
   0: rust_begin_unwind
             at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library/std/src/panicking.rs:652:5
   1: core::panicking::panic_fmt
             at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library/core/src/panicking.rs:72:14
   2: core::result::unwrap_failed
             at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library/core/src/result.rs:1679:5
   3: core::result::Result<T,E>::unwrap
             at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library/core/src/result.rs:1102:23
   4: risingwave_meta_node::server::start_service_as_election_leader::{{closure}}
             at ./src/meta/node/src/server.rs:459:27
   5: risingwave_meta_node::server::rpc_serve_with_store::{{closure}}
             at ./src/meta/node/src/server.rs:333:6
   6: risingwave_meta_node::server::rpc_serve::{{closure}}
             at ./src/meta/node/src/server.rs:184:14
   7: risingwave_meta_node::start::{{closure}}
             at ./src/meta/node/src/lib.rs:465:10
   8: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library/core/src/future/future.rs:123:9
   9: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library/core/src/future/future.rs:123:9
  10: <&mut F as core::future::future::Future>::poll
             at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library/core/src/future/future.rs:111:9
  11: risingwave_rt::main_okk::{{closure}}::{{closure}}
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823-shallow/tokio-1.38.0/src/macros/select.rs:548:49
  12: <tokio::future::poll_fn::PollFn<F> as core::future::future::Future>::poll
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823-shallow/tokio-1.38.0/src/future/poll_fn.rs:58:9
  13: risingwave_rt::main_okk::{{closure}}
             at ./src/utils/runtime/src/lib.rs:107:9
  14: <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823-shallow/tracing-0.1.40/src/instrument.rs:321:9
  15: tokio::runtime::park::CachedParkThread::block_on::{{closure}}
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823-shallow/tokio-1.38.0/src/runtime/park.rs:281:63
  16: tokio::runtime::coop::with_budget
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823-shallow/tokio-1.38.0/src/runtime/coop.rs:107:5
  17: tokio::runtime::coop::budget
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823-shallow/tokio-1.38.0/src/runtime/coop.rs:73:5
  18: tokio::runtime::park::CachedParkThread::block_on
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823-shallow/tokio-1.38.0/src/runtime/park.rs:281:31
  19: tokio::runtime::context::blocking::BlockingRegionGuard::block_on
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823-shallow/tokio-1.38.0/src/runtime/context/blocking.rs:66:9
  20: tokio::runtime::scheduler::multi_thread::MultiThread::block_on::{{closure}}
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823-shallow/tokio-1.38.0/src/runtime/scheduler/multi_thread/mod.rs:87:13
  21: tokio::runtime::context::runtime::enter_runtime
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823-shallow/tokio-1.38.0/src/runtime/context/runtime.rs:65:16
  22: tokio::runtime::scheduler::multi_thread::MultiThread::block_on
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823-shallow/tokio-1.38.0/src/runtime/scheduler/multi_thread/mod.rs:86:9
  23: tokio::runtime::runtime::Runtime::block_on
             at /Users/dylan/.cargo/registry/src/github.com-1ecc6299db9ec823-shallow/tokio-1.38.0/src/runtime/runtime.rs:349:45
  24: risingwave_rt::main_okk
             at ./src/utils/runtime/src/lib.rs:149:5
  25: risingwave_cmd::meta
             at ./src/cmd/src/lib.rs:47:5
  26: risingwave::Component::start
             at ./src/cmd_all/src/bin/risingwave.rs:121:27
  27: risingwave::main
             at ./src/cmd_all/src/bin/risingwave.rs:224:5
  28: core::ops::function::FnOnce::call_once
             at /rustc/72fdf913c53dd0e75313ba83e4aa80df3f6e2871/library/core/src/ops/function.rs:250:5
```

File: Cargo.lock [hunk 2]
```diff
@@ -1482,12 +1488,11 @@ dependencies = [
 
 [[package]]
 name = "aws-sdk-s3"
-version = "1.1.0"
+version = "1.17.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "62d240c8c3e3663cd278e47702bbd2566203362d93b51d95575d7b2e0c265e99"
+checksum = "93d35d39379445970fc3e4ddf7559fff2c32935ce0b279f9cb27080d6b7c6d94"
```
Comment (chenzl25) [inline Cargo.lock:1493]: `aws-sdk-s3` will be upgraded to `1.17.0`

**Comment by chenzl25** [comment]
Longevity result: 
https://grafana.test.risingwave-cloud.xyz/d/EpkBw5W4k/risingwave-dev-dashboard?orgId=1&var-datasource=P2453400D1763B4D9&var-namespace=usrlngvty-20240820-145456&var-instance=benchmark-risingwave&var-pod=All&var-component=All&var-table=All&from=1724167211231&to=1724227936440

https://buildkite.com/risingwave-test/longevity-test/builds/1785#01917047-d891-4ae2-a827-d7c27cf895f0

**Comment by chenzl25** [comment]
nexmark performance https://buildkite.com/risingwave-test/nexmark-benchmark/builds/4395#01917405-6be2-4161-ba43-06e5b42be07d

**Comment by hzxa21** [comment]
image

There seems to be a drop in q0 and q7 performance.

**Comment by chenzl25** [comment]
> image
> 
> There seems to be a drop in q0 and q7 performance.

Do you think we can merge this PR?

**Comment by chenzl25** [comment]
As I know, this PR https://github.com/risingwavelabs/risingwave/pull/18011 has been merged which means we will use opendal by default for object store.

**Comment by chenzl25** [comment]
The performance result looks good. Let's merge this PR.
image

**Pull request merged**